src/plugins/intel_gpu/src/graph/graph_optimizer/remove_redundant_reorders.cpp

```diff
@@ -284,24 +284,29 @@ void remove_redundant_reorders::run(program& p) {
             i_layout.data_padding.upper_size().spatial[1] == 0 && i_layout.data_padding.lower_size().spatial[1] == 0 &&
             o_layout.data_padding.upper_size() == (tensor)0 && o_layout.data_padding.lower_size() == (tensor)0 &&
             i_layout.data_type == o_layout.data_type) {
-            r_node.can_be_optimized(true);
-            r_node.requires_reinterpret(true);
+            // If the newly aligned pad is merged into output layout during post_optimize_graph phase
+            // and then buffer is reinterpreted, user node cannot handle pad properly for kernel execution
+            if (!update_implementations || (i_layout.feature() % 16 == 0 &&
+                i_layout.data_padding == padding() && o_layout.data_padding == padding())) {
```

Comment:
Sorry but could you add one more condition as "batch size == 0" ? 

```
 if (!update_implementations || (i_layout.feature() % 16 == 0 &&
                i_layout.data_padding == padding() && o_layout.data_padding == padding())) || i_layout.batch() == 1) {
```

Reply:
updated